src/theory/quantifiers/sygus_sampler.cpp

```diff
@@ -678,12 +679,12 @@ Node SygusSampler::getSygusRandomValue(TypeNode tn,
       }
       Trace("sygus-sample-grammar")
           << "  child #" << i << " : " << c << std::endl;
-      pre[i] = c;
+      children.emplace_back(c);
     }
     if (success)
     {
       Trace("sygus-sample-grammar") << "mkGeneric" << std::endl;
```

Comment:
Using 'mkSygusTerm' in the trace message may be less misleading than keeping 'mkGeneric'.